modules/java-modules/elastic-v2/src/main/java/org/syslog_ng/elasticsearch_v2/messageprocessor/http/HttpBulkMessageProcessor.java

```diff
@@ -69,7 +78,31 @@ public boolean send(Index index) {
 			flush();
 		}
 		bulk = bulk.addAction(index);
+		lastSentTime = getCurrentUnixEpochSecond();
 		messageCounter++;
 		return true;
 	}
+
+    private long getCurrentUnixEpochSecond() {
+        return System.currentTimeMillis() / 1000L;
+    }
+
+    private void startFlushTimer() {
+        timer = new Timer();
+        timer.scheduleAtFixedRate(new TimerTask() {
```

Comment:
This makes the flush-timeout unpredictable, isn't it? This task is executed each second periodically. In case of the last message arrived in the early part of the period, flush would take almost an extra second to be triggered. Can this be a problem? Probably rescheduling a task after each sent message would be more deterministic.

Reply:
it means the flush will happen between flush_timout and flush_timeout+1. I don't think this can be a problem.

The problem is that in java there is no way to reschedule a task. You have to cancel it then recreate. Now if syslog-ng processes 20k msg/sec, you should do the timer cancel+recreate 20k/sec that is waste of resources (everything is very expensive that we do per message).

My solution will spare the resources against a theoretically very low delaying.